manifests/init.pp

```diff
@@ -50,22 +50,30 @@
 #  If set to `['never']`, then this module will not create any incidents at all.
 class servicenow_reporting_integration (
   String[1] $instance,
-  String[1] $caller_id,
-  Optional[String[1]] $pe_console_url                = undef,
-  String $servicenow_credentials_validation_table    = 'incident',
-  Optional[String[1]] $user                          = undef,
-  Optional[String[1]] $password                      = undef,
-  Optional[String[1]] $oauth_token                   = undef,
-  Optional[String[1]] $category                      = undef,
-  Optional[String[1]] $subcategory                   = undef,
-  Optional[String[1]] $contact_type                  = undef,
-  Optional[Integer] $state                           = undef,
-  Optional[Integer] $impact                          = undef,
-  Optional[Integer] $urgency                         = undef,
-  Optional[String[1]] $assignment_group              = undef,
-  Optional[String[1]] $assigned_to                   = undef,
+  Optional[String[1]] $user                                                                  = undef,
+  Optional[String[1]] $password                                                              = undef,
+  Optional[String[1]] $oauth_token                                                           = undef,
+  Enum['event_management', 'incident_management'] $operation_mode                            = 'event_management',
+  Optional[String[1]] $caller_id                                                             = undef,
+  Optional[String[1]] $pe_console_url                                                        = undef,
+  Optional[String[1]] $category                                                              = undef,
+  Optional[String[1]] $subcategory                                                           = undef,
+  Optional[String[1]] $contact_type                                                          = undef,
+  Optional[Integer] $state                                                                   = undef,
+  Optional[Integer] $impact                                                                  = undef,
+  Optional[Integer] $urgency                                                                 = undef,
+  Optional[String[1]] $assignment_group                                                      = undef,
+  Optional[String[1]] $assigned_to                                                           = undef,
   Servicenow_reporting_integration::IncidentCreationConditions $incident_creation_conditions = ['failures', 'corrective_changes'],
+  Optional[String[1]] $servicenow_credentials_validation_table                               = undef,
 ) {
+  if $operation_mode == 'incident_management' {
```

Comment:
This, for example, is nicer to read with `operation_mode` instead of a comment like `// user turned off event management so we default to incident. caller_id is required param for that case` ...